content/300-guides/050-database/850-using-prisma-with-planetscale.mdx

```diff
@@ -0,0 +1,240 @@
+
+- **Referential integrity.** To support scaling across multiple database servers, PlanetScale [does not allow the use of foreign key constraints](https://docs.planetscale.com/tutorials/operating-without-foreign-keys), which are normally used in relational databases to enforce relationships between data in different tables. To maintain these relationships in your data while using PlanetScale with Prisma, you will need to use Prisma's ability to [emulate referential integrity in the Prisma client](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-integrity#handling-the-referential-integrity-in-prisma). For more information, see [How to enable emulation of referential integrity](#how-to-enable-emulation-of-referential-integrity).
+
+- **Branching and deploy requests.** PlanetScale provides two types of database branches: **development branches**, which allow you to test out schema changes, and **production branches**, which are protected from direct schema changes. Instead, changes must be first created on a development branch and then deployed to production using a deploy request. Production branches are highly available and include automated daily backups. To learn more, see [How to use branches and deploy requests](#how-to-use-branches-and-deploy-requests).
+
+- **Creating indexes on foreign keys.** In a standard MySQL database, if a table has a column with a foreign key constraint, an index is automatically created on that column. As PlanetScale does not support foreign keys, these indexes are not created which can lead to issues with queries not being well optimised. To avoid this, you can create indexes in Prisma. For more information, see [How to create indexes on foreign keys](#how-to-create-indexes-on-foreign-keys).
+
+- **Making schema changes with `db push`.** When you merge a development branch into your production branch, PlanetScale will automatically compare the two schemas and generate its own schema diff. This means that Prisma's [`prisma migrate`](/concepts/components/prisma-migrate) workflow, which generates its own history of migration files, is not a natural fit when working with PlanetScale. These migration files may not reflect the actual schema changes run by PlanetScale when the branch is merged.
+
+  <Admonition type="warning">
+
+  Instead, Prisma recommends you use the `prisma db push` command when making
+  schema changes with PlanetScale.
+
+  </Admonition>
+
+  For an example of how this works, see [How to make schema changes with `db push`](#how-to-make-schema-changes-with-db-push)
+
+- **Introspection**. When you introspect on an existing database, you will get a schema with no relations, as they are usually defined based on foreign keys that connect tables. You will need to add the missing relations in manually. For more information, see [How to add in missing relations after Introspection](#how-to-add-in-missing-relations-after-Introspection).
```

Comment:
Potential clarification:
```suggestion
- **Introspection**. When you introspect on an existing database, you will get a schema with no relations, as they are usually defined based on foreign keys that connect tables. Because PlanetScale does not support foreign keys, and you use Prisma to emulate referential integrity, you will need to add the missing relations in manually. For more information, see [How to add in missing relations after Introspection](#how-to-add-in-missing-relations-after-Introspection).
```